docs/labs/lab06-provisioning/vc_task_retrigger.py

```diff
@@ -4,39 +4,109 @@
 
-# Create connection to CloudVision
-clnt = CvpClient()
-
-clnt.connect(nodes=['cvp1'], username="username",password="password")
-
-# Trigger tasks after they were cancelled
-
-devices = ["tp-avd-leaf1", "tp-avd-leaf2", "tp-avd-leaf3", "tp-avd-leaf4", "tp-avd-spine1", "tp-avd-spine2"]
-
-compliance = {"0001": "Config is out of sync",
-              "0003": "Config & image out of sync",
-              "0004": "Config, Image and Device time are in sync",
-              "0005": "Device is not reachable",
-              "0008": "Config, Image and Extensions are out of sync",
-              "0009": "Config and Extensions are out of sync",
-              "0012": "Config, Image, Extension and Device time are out of sync",
-              "0013": "Config, Image and Device time are out of sync",
-              "0014": "Config, Extensions and Device time are out of sync",
-              "0016": "Config and Device time are out of sync",
-}
-
-for dev in devices:
-   # generate device object
-   device = clnt.api.get_device_by_name(dev)
-   # generate configlet list
-   cl = clnt.api.get_configlets_by_device_id(device['systemMacAddress'])
-   # generate a task if config is out of sync
-   if device['complianceCode'] in compliance.keys():
-      print(clnt.api.apply_configlets_to_device("", device, cl))
+
+
+
+def main():
+        
+    compliance = {"0001": "Config is out of sync",
+        "0003": "Config & image out of sync",
+#        "0004": "Config, Image and Device time are in sync",
+        "0005": "Device is not reachable",
+        "0008": "Config, Image and Extensions are out of sync",
+        "0009": "Config and Extensions are out of sync",
+        "0012": "Config, Image, Extension and Device time are out of sync",
+        "0013": "Config, Image and Device time are out of sync",
+        "0014": "Config, Extensions and Device time are out of sync",
+        "0016": "Config and Device time are out of sync"
+    }
+    # Create connection to CloudVision
+    clnt = CvpClient()
+
+
+    parser = argparse.ArgumentParser(description='Script to recreate a task, if a previous config push was cancelled')
+    parser.add_argument('-u', '--username', default='username')
+    parser.add_argument('-p', '--password', default=None)
+    parser.add_argument('-c', '--cvpserver', action='append')
+    parser.add_argument('-f', '--filter', action='append', default=None)
+    args = parser.parse_args()
+
+
+    if args.password is None:
+        args.password = getpass()
+
+    for cvpserver in args.cvpserver:
+        print("Connecting to %s" % cvpserver)
+        try:
+            clnt.connect(nodes=[cvpserver], username=args.username, password=args.password)
+        except Exception as e:
+            print("Unable to connect to CVP: %s" % str(e))
+        #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        # Trigger compliance check here
+        #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1
+        try:
+            clnt.api.check_compliance('root', 'container')
```

Comment:
Maybe we should add a note here? as we don't need to do this post 2020.3.0 where we have continous compliance check, so running `get_device_by_name(device)` or `get_inventory()` will get you always the latest compliance codes

Reply:
Or just add a check - only do this if the CVP release is <2020.3.0

Reply:
good idea! you could use `get_cvp_info()`

Reply:
@colinmacgiolla are you planning to add the version check before I merge this?